rpcserver.go

```diff
@@ -491,26 +491,30 @@ func (r *rpcServer) MintAsset(ctx context.Context,
 	}
 }
 
-// checkFeeRateSanity ensures that the provided fee rate is above the same
-// minimum fee used as a floor in the fee estimator.
+// checkFeeRateSanity ensures that the provided fee rate, in sat/vB, is above
+// the same minimum fee used as a floor in the fee estimator.
 func checkFeeRateSanity(rpcFeeRate uint32) (*chainfee.SatPerKWeight, error) {
-	var feeRate *chainfee.SatPerKWeight
-	switch {
+	feeFloor := chainfee.FeePerKwFloor.FeePerKVByte()
+
 	// No manual fee rate was set, which is the default.
-	case rpcFeeRate == 0:
+	if rpcFeeRate == 0 {
+		return nil, nil
+	}
 
+	// Cast the fee rate from sat/vB to sat/kb below comparing against our
+	// fee floor.
+	manualFeeRate := chainfee.SatPerKVByte(rpcFeeRate * 1000)
+	switch {
 	// A manual fee was set but is below a reasonable floor.
-	case rpcFeeRate < uint32(chainfee.FeePerKwFloor):
-		return nil, fmt.Errorf("manual fee rate %d below floor of %d",
-			rpcFeeRate, uint32(chainfee.FeePerKwFloor))
+	case manualFeeRate < feeFloor:
+		return nil, fmt.Errorf("manual fee rate %s below floor of %s",
```

Comment:
I think it's okay to round up to the floor rate here instead of returning an error.

Reply:
I'm not sure if I agree with you Oli. If the fee rate was set manually maybe the user strictly didn't want to pay a higher fee. I think we should add a flag to the relevant RPC calls if we want to automatically round up. Could be a future PR.

If this error does remain, then I think we should roll it an `error` type and then check for it in the itest.

Reply:
The rounding up is a difference of 3 sat/kw or 12 sat/kvB; so extremely small, even for a hypothetical large transfer TX.

This is also what we do in Pool:

https://github.com/lightninglabs/pool/blob/59c7c2b40009503851de6da3b3f9b4d90b34e208/cmd/pool/account.go#L172

Without the rounding, the TX would get rejected later on by LND before broadcast anyways, so the alternative (no rounding) is not allowing the user to specify a fee below 2 sat/vB.

I think the suggestion is to round on the CLI where the user cannot specify something above the fee floor but below 2 sat/vB, and then return an error on the actual RPC interface where a user can specify sat/kw directly.

Reply:
The change from 250 to 253 as the floor fee rate was a security fix we did in `lnd`. So devs don't necessarily know about it, and we do them a favor by doing it that way, otherwise their TX could get stuck.
And as mentioned above, we do this in all our projects. Basically you should never publish a TX with just 250 sat/kWU.

Reply:
Confirmed manually that the CLI rounding works.